### DISCUSSION TITLE
Overlay scroll to top button requirement

### DISCUSSION THREAD

**Discussion by jaymehta-twilio** [twilio-labs/paste#1518] [Q&A] [answered]
Hello team,

We have a requirement to provide a `scroll to top` button for one of the page. Are there any guidelines or overlay provision in today's Paste component ?

<img width="1242" alt="Screen Shot 2021-06-02 at 11 29 29 AM" src="https://user-images.githubusercontent.com/52717786/120533849-07e6a980-c396-11eb-9d82-e72d583a2430.png">

**Comment by zahnster** [comment]
Hi @jaymehta1986,

Can you provide some additional context? Is it that you are hoping the 'Back to top' button will be affixed to the viewport (ala, position: fixed) when the user scrolls down?

**Reply by jaymehta-twilio** [reply]
yes, the screenshot is based off of the designs but that's the goal. It'd be great to be rendered within an overlay.

**Comment by richbachman** [comment, marked as answer]
Hi @jaymehta1986 

Following up on this one. After discussing with the team, we don't think we will be providing a component for this since this more of a layout consideration.

That said, I think you could accomplish something like this with Paste components. You could place the Button in a fixed position Box, so it always stays at the bottom right of the viewport. You'll mainly need to be careful about what content the button might cover up. You could tie into the scroll position to change the opacity of that Button, so it only shows up based on scroll.

Let us know if that helps, and if you come up with a solution using Paste we'd like to see that as well.

**Reply by jaymehta-twilio** [reply]
Thanks @richbachman. I think this shouldn't be complicated. Will update once we have an acceptable solution.

**Reply by jaymehta-twilio** [reply]
Hey @richbachman, following up on this thread. We've implemented a `Back to top` button [here](https://deploy-preview-182.console.dev.twilio.com/us1/develop/event-streams/subscriptions/new). Please scroll a little for it to appear. It works great. The only question I had was that since it's a fixed positioned button, the accessibility could be an issue. Meaning, the button may come last in natural tab flow. I was wondering if there're any concerns/inputs for those.

<img width="1157" alt="Screen Shot 2021-07-09 at 11 30 20 AM" src="https://user-images.githubusercontent.com/52717786/125343242-b1f91000-e30a-11eb-8042-ce7d0cbbf292.png">

**Reply by gloriliale** [reply]
Hey @jaymehta-twilio, nice! This is awesome. About the accessibility of the button: it depends on if this is a static list, or if things are added to the list as you scroll, like an infinite list. Having it at the bottom of a static list, where everything is present when you render, would be fine. It'll be trickier if that's not the case.

**Reply by jaymehta-twilio** [reply]
Got it, thanks. Yes it's a static list and won't grow on scroll. So, what I understand from above comment is that it's fine if the scroll button comes last in the tab order after all the components from the static list. Please correct me if I'm wrong. Thank you again :).

**Reply by gloriliale** [reply]
That's right!
